### ISSUE TITLE
Crash on Mac M1 when creating schema containing embeddings field.

### ISSUE THREAD

**Issue by polkabrothers** [typesense/typesense#1171]
## Description

Typesense crashes when trying to create schema containing embeddings field, and also destroys previous existing collections.
I'm using the example from the API documentation.
Typesense 0.25.0 installed through brew, but I also tried downloading the binary and starting up the server that way, same result.
Tested in both PHP and Python.

Also tried to clear the data-dir and renaming it but still get a crash.
The models/ directory just contains the config.json and nothing more, tried with different models but the same result.
If I remove the "embedding" array so that the schema is created without embeddings it works as normal with no crash.

I have provided a file with the error output.

## Steps to reproduce

`use Typesense\Client;

$clientConfig = [
    'api_key' => '12345',
    'nodes' => [
        [
            'host' => 'localhost',
            'port' => '8108',
            'protocol' => 'http'
        ],
    ],
    'connection_timeout_seconds' => 2
];

$client = new Client($clientConfig);

$schema = [
  "name" => "products",
  "fields" => [
    [
      "name" => "brand",
      "type" => "string"
    ],
    [
      "name" => "categories",
      "type" => "string[]"
    ],
    [
      "name" => "embedding",
      "type" => "float[]",
      "embed" => [
        "from" => [
          "brand",
          "categories"
        ],
        "model_config" => [
          "model_name" => "ts/all-MiniLM-L12-v2"
        ]
      ]
    ]
  ]
];

$client->collections->create($schema);`

[ts_v0.25.0.txt](https://github.com/typesense/typesense/files/12388522/ts_v0.25.0.txt)
## Metadata

0.25.0

**OS**:

Mac Mini M1, 8gb
macOS Ventura 13.5.1

**Comment by jasonbosco** [comment]
@polkabrothers Could you try running Typesense via Docker to see if the issue persists there as well?

**Comment by polkabrothers** [comment]
Tried in Docker and it works without problem, I don't use usually use Docker in dev/prod though so It's still a problem though.

**Comment by jasonbosco** [comment]
Thank you for confirming. We ran into an API compatibility issue with macOS M1 with v0.25 last minute, which would have required some non-trivial amount of refactoring to get it to work. 

So for 0.25, we decided to drop native-binary support for macOS running on M1/M2 and recommend usage of Typesense via Docker for macOS on M1/M2, especially because we now publish native Linux ARM builds which will work well with Docker on M1/M2. 

We plan to address macOS native binary compatibility with M1/M2 in future releases.

**Comment by tahola** [comment]
I have the same issue, did you find a solution without Docker @polkabrothers ? My typesense keeps trying to download the model (I think) and keep crashing, even if I had remove the collection.

**Comment by AndyHoang** [comment]
> Tried in Docker and it works without problem, I don't use usually use Docker in dev/prod though so It's still a problem though.

Can you tell the docker tag that you' had used?

I tried the latest docker tag and it seem to have the problem that you described.

> With embed field with any models, the schema creation process succeed but the container max its cpu/ram after trigger adding one record.
> Remove the embed field and thing work like normal.

```
        'fields': [
            { 'name': "id", 'type': "string"},
            { 'name': "by", 'type': "string", 'facet': True },
            { 'name': "time", 'type': "int64" },
            { 'name': "text", 'type': "string" },
            { 'name': "embedding",
                'type': "float[]",
                'embed': {
                    'from': ["text"],
                    'model_config': {
                        "model_name": "ts/e5-small"
                    }
                }
            },
            { 'name': "parent", 'type': "string", 'optional': True },
        ],
```
My field in my schema, copy from this [example](https://github.com/typesense/showcase-hn-comments-semantic-search)

**Comment by tahola** [comment]
@AndyHoang for me this one is working well **typesense/typesense:0.25.1.rc6**, I had an issue with the [0.25.0](https://github.com/typesense/typesense/issues/1206)

**Comment by jasonbosco** [comment]
We've now published `0.25.1` which should resolve this issue.
